MixinGuiResearchBrowser cleanup, additional creative functionality

Midnight145 opened the pull request:
**What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
This cleans up MixinGuiResearchBrowser and separates it into three different mixins instead of one, so we don't have a bunch of extra methods mixed in with a `if (option.isEnabled())` checks everywhere.

Additionally, while in creative mode, you no longer need scribing tools and paper to get research notes.

**What is the current behavior?** (You can also link to an open issue here)
All four features (creative functionality, ctrl scroll, research ids, and right click navigation) are all currently in the same file so we get a bunch of unnecessary injections if only a subset of those four are enabled.

Additionally, to get research notes in creative, you are still required to have paper and scribing tools.

**What is the new behavior (if this is a feature change)?**
Separates the four mixins into three files which helps avoid any potential unnecessary injections, and makes scribing tools/paper no longer necessary in creative.

**Does this PR introduce a breaking change?**
No

**Other information**:
Creative functionality and ctrl + scroll functionality are still in the same mixin due to them both needing to override handleInput and there was no clean enough solution for that that I was happy with.

Nikolay-Sitnikov commented:
If you're in Creative Mode, you can always Ctrl-click to research something for free, so I am not sure how impactful the research note change is.

Midnight145 commented:
fixed

rndmorris commented on src/main/java/dev/rndmorris/salisarcana/lib/MixinHelpers.java:
```diff
@@ -0,0 +1,38 @@
+
+    // all the BrowserPaging reflection is temporary until I get a PR to tc4tweaks
+
+    private static final BrowserPagingAPI browserPaging = TC4TweaksAPI.getBrowserPagingAPI();
```
As discussed in DMs, this causes a crash if the mod is installed alongside an older version of tc4tweaks.

rndmorris reviewed (changes requested):
Tested:
- [x] Creative research note
- [x] Creative research completion
- [x] Category page scrolling
- [x] Display research key on ctrl-hover

Issues:
* Holding a Ctrl-click on reserach when you don't have the prerequisites spams your chat with "You lack the required knowledge to understand this."
* Category page scrolling crashes if used with an older version of TC4Tweaks.

Midnight145 commented:
> Holding a Ctrl-click on reserach when you don't have the prerequisites spams your chat with "You lack the required knowledge to understand this."

fixed
> Category page scrolling crashes if used with an older version of TC4Tweaks.

fixed by requiring new version of tc4tweak in dependency string

rndmorris reviewed (approved):
Tested as fixed:
- [x] Holding a Ctrl-click on reserach when you don't have the prerequisites spams your chat with "You lack the required knowledge to understand this."
  - No more spam
- [x] Category page scrolling crashes if used with an older version of TC4Tweaks.
  - Version checking works as expected, prevents installing an incompatible tc4tweaks version.
